revm-utils/src/evm.rs

```diff
@@ -0,0 +1,228 @@
+            .unwrap();
+
+        let spec_id = get_spec_id_for_block(
+            &self.network,
+            &self.fork_schedule,
+            block.header().timestamp(),
+        );
+        let spec_id = spec_id.unwrap_or(SpecId::LATEST);
+        let handler_cfg = HandlerCfg {
+            spec_id,
+            is_optimism: false,
+        };
+
+        let mut cfg = CfgEnv::default();
+        cfg.chain_id = self.chain_id;
+        cfg.disable_block_gas_limit = !validate_tx;
+        cfg.disable_eip3607 = !validate_tx;
+        cfg.disable_base_fee = !validate_tx;
+
+        let env = Env {
+            tx: Self::tx_env(tx),
+            block: Self::block_env(&block, &self.fork_schedule),
+            cfg,
+        };
+
+        EnvWithHandlerCfg::new(Box::new(env), handler_cfg)
+    }
+
+    fn tx_env(tx: &N::TransactionRequest) -> TxEnv {
+        // TODO: Handle the Tx fields based on networks
+        TxEnv {
+            caller: tx.from().unwrap_or_default(),
+            gas_limit: tx.gas_limit().unwrap_or(u64::MAX),
+            gas_price: tx.gas_price().map(U256::from).unwrap_or_default(),
+            transact_to: tx.kind().unwrap_or_default(),
+            value: tx.value().unwrap_or_default(),
+            data: tx.input().unwrap_or_default().clone(),
+            nonce: tx.nonce(),
+            chain_id: tx.chain_id(),
+            access_list: tx.access_list().map(|v| v.to_vec()).unwrap_or_default(),
+            gas_priority_fee: tx.max_priority_fee_per_gas().map(U256::from),
+            max_fee_per_blob_gas: None,
+            //max_fee_per_blob_gas = tx.max_fee_per_blob_gas().map(U256::from);
+            blob_hashes: Vec::new(),
+            //blob_hashes = tx
+            //     .blob_versioned_hashes()
+            //     .as_ref()
+            //     .map(|v| v.to_vec())
+            //     .unwrap_or_default();
+            authorization_list: None,
+            optimism: OptimismFields::default(),
+        }
+    }
+
+    fn block_env(block: &N::BlockResponse, fork_schedule: &ForkSchedule) -> BlockEnv {
+        // Check only required in Certain network cases
+        let is_prague = block.header().timestamp() >= fork_schedule.prague_timestamp;
+        let blob_excess_gas_and_price = block
+            .header()
+            .excess_blob_gas()
+            .map(|v| BlobExcessGasAndPrice::new(v, is_prague))
+            .unwrap_or_else(|| BlobExcessGasAndPrice::new(0, is_prague));
+
+        BlockEnv {
+            number: U256::from(block.header().number()),
+            coinbase: block.header().beneficiary(),
+            timestamp: U256::from(block.header().timestamp()),
+            gas_limit: U256::from(block.header().gas_limit()),
+            basefee: U256::from(block.header().base_fee_per_gas().unwrap_or(0_u64)),
+            difficulty: block.header().difficulty(),
+            prevrandao: block.header().mix_hash(),
+            blob_excess_gas_and_price: Some(blob_excess_gas_and_price),
```

Comment:
How does this interact OP networks? Should this be set to `None` in that case?

Reply:
it will be `BlobExcessGasAndPrice::new(0, is_prague)`